packages/terra-list/src/SelectableList.jsx

```diff
@@ -38,36 +43,76 @@ const defaultProps = {
   disableUnselectedItems: false,
   isDivided: false,
   hasChevrons: false,
+  hasSections: false,
   onChange: undefined,
   selectedIndexes: [],
 };
 
-const SelectableList = ({
-  children,
-  disableUnselectedItems,
-  isDivided,
-  onChange,
-  hasChevrons,
-  selectedIndexes,
-  ...customProps
-}) => {
-  const clonedChildren = React.Children.map(children, (child, index) => {
-    const wrappedOnClick = SelectableUtils.wrappedOnClickForItem(child, index, onChange);
-    const wrappedOnKeyDown = SelectableUtils.wrappedOnKeyDownForItem(child, index, onChange);
-    const newProps = SelectableUtils.newPropsForItem(child, index, wrappedOnClick, wrappedOnKeyDown, hasChevrons, selectedIndexes, disableUnselectedItems);
-    return React.cloneElement(child, newProps);
-  });
+class SelectableList extends React.Component {
+  static getDerivedStateFromProps(props, state) {
+    if (props.hasSections !== state.hasSections) {
+      return {
+        hasSections: props.hasSections,
+      };
+    }
+    return null;
+  }
 
-  return (
-    <List isDivided={isDivided} role="listbox" {...customProps}>
-      {clonedChildren}
-    </List>
-  );
-};
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasSections: false,
```

Comment:
Again, why is hasSections part of state?

Reply:
Addressed above.